File: 1430-find-the-k-beauty-of-a-number/1430-find-the-k-beauty-of-a-number.java
class Solution {
    public int divisorSubstrings(int num, int k) {
        StringBuilder s= new StringBuilder("");
        int result = 0;
        s.append(num);
        for (int i = 0; i < s.length() - k + 1; i++ ) {
            String t = s.substring(i, i + k);
            int x = Integer.parseInt(t);
            if (x != 0 && num% x == 0) {
                result++;
            }
        }
        return result;

        
    }
}

File: 1430-find-the-k-beauty-of-a-number/1430-find-the-k-beauty-of-a-number.py
class Solution:
    def divisorSubstrings(self, num: int, k: int) -> int:
        s = str(num)

        count = 0
        if len(s) == 1:
            return 1
        for i in range(len(s) - k + 1):
            x = int(s[i: i + k])

            if (x != 0 and num % x == 0):
                count += 1

        return count









        